src/main/java/org/ambraproject/rhino/model/article/ArticleMetadata.java

```diff
@@ -311,39 +322,42 @@ public boolean equals(Object o) {
     if (doi != null ? !doi.equals(that.doi) : that.doi != null) return false;
     if (title != null ? !title.equals(that.title) : that.title != null) return false;
     if (eIssn != null ? !eIssn.equals(that.eIssn) : that.eIssn != null) return false;
-    if (description != null ? !description.equals(that.description) : that.description != null) return false;
+    if (journalName != null ? !journalName.equals(that.journalName) : that.journalName != null)
+      return false;
+    if (description != null ? !description.equals(that.description) : that.description != null)
+      return false;
     if (rights != null ? !rights.equals(that.rights) : that.rights != null) return false;
     if (language != null ? !language.equals(that.language) : that.language != null) return false;
     if (format != null ? !format.equals(that.format) : that.format != null) return false;
-    if (pageCount != null ? !pageCount.equals(that.pageCount) : that.pageCount != null) return false;
-    if (eLocationId != null ? !eLocationId.equals(that.eLocationId) : that.eLocationId != null) return false;
-    if (publicationDate != null ? !publicationDate.equals(that.publicationDate) : that.publicationDate != null) {
+    if (pageCount != null ? !pageCount.equals(that.pageCount) : that.pageCount != null)
+      return false;
+    if (eLocationId != null ? !eLocationId.equals(that.eLocationId) : that.eLocationId != null)
+      return false;
+    if (publicationDate != null ? !publicationDate.equals(that.publicationDate) : that.publicationDate != null)
       return false;
-    }
     if (volume != null ? !volume.equals(that.volume) : that.volume != null) return false;
     if (issue != null ? !issue.equals(that.issue) : that.issue != null) return false;
-    if (publisherLocation != null ? !publisherLocation.equals(that.publisherLocation) : that.publisherLocation != null) {
+    if (publisherLocation != null ? !publisherLocation.equals(that.publisherLocation) : that.publisherLocation != null)
+      return false;
+    if (publisherName != null ? !publisherName.equals(that.publisherName) : that.publisherName != null)
       return false;
-    }
-    if (publisherName != null ? !publisherName.equals(that.publisherName) : that.publisherName != null) return false;
     if (url != null ? !url.equals(that.url) : that.url != null) return false;
-    if (nlmArticleType != null ? !nlmArticleType.equals(that.nlmArticleType) : that.nlmArticleType != null) {
+    if (nlmArticleType != null ? !nlmArticleType.equals(that.nlmArticleType) : that.nlmArticleType != null)
```

Comment:
+1 to one of the most creative use of the **ternary** operator, haven't seen it in an if-statement.  I guess it boils down to:

```
if (nlmArticleType != null && !nlmArticleType.equals(that.nlmArticleType)) {
 return false;
}
```
Not necessary to have **that.nlmArticleType != null**, as **!nlmArticleType.equals(that.nlmArticleType)** will handle that case.

Reply:
This code is all auto-generated by IntelliJ so I can't claim any credit :)
Yes, it is unoptimized, but we typically don't mess with the auto-generated equals/hashcode cause we consider it "under the covers" so to speak.